Lib/test/test_builtin.py

```diff
@@ -2372,24 +2372,31 @@ def __del__(self):
 
 @cpython_only
 class ImmortalTests(unittest.TestCase):
-    def test_immortal(self):
-        none_refcount = sys.getrefcount(None)
-        true_refcount = sys.getrefcount(True)
-        false_refcount = sys.getrefcount(False)
-        smallint_refcount = sys.getrefcount(100)
-
-        # Assert that all of these immortal instances have large ref counts.
-        self.assertGreater(none_refcount, 2 ** 15)
-        self.assertGreater(true_refcount, 2 ** 15)
-        self.assertGreater(false_refcount, 2 ** 15)
-        self.assertGreater(smallint_refcount, 2 ** 15)
-
-        # Confirm that the refcount doesn't change even with a new ref to them.
-        l = [None, True, False, 100]
-        self.assertEqual(sys.getrefcount(None), none_refcount)
-        self.assertEqual(sys.getrefcount(True), true_refcount)
-        self.assertEqual(sys.getrefcount(False), false_refcount)
-        self.assertEqual(sys.getrefcount(100), smallint_refcount)
+
+    if sys.maxsize < (1 << 32):
+        IMMORTAL_REFCOUNT = (1 << 30) - 1
+    else:
+        IMMORTAL_REFCOUNT = (1 << 32) - 1
+
+    IMMORTALS = (None, True, False, Ellipsis, NotImplemented, *range(-5, 257))
+
+    def assert_immortal(self, immortal):
+        with self.subTest(immortal):
+            self.assertEqual(sys.getrefcount(immortal), self.IMMORTAL_REFCOUNT)
+
+    def test_immortals(self):
+        for immortal in self.IMMORTALS:
```

Comment:
Small nit: I generally prefer rolling out assertions in unit tests to make it easier to isolate by line-number in the place where the test failed. But the practice is not standardized so I'm fine either way

Reply:
Yeah, I'd like comprehensive testing of these immortals though, and I don't want to roll out ~250 different assertions for each test.

I figured making each one a `subTest` in `assert_immortal` is a pretty good compromise.